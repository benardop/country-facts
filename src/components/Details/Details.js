import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import style from './Details.module.css';
import { deleteDetails } from '../../redux/detailsReducer/detailsReducer';
import arrowImg from '../../assets/images/arrow.svg';

function Details() {
  const details = useSelector((state) => state.details);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(deleteDetails());
    window.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
  }, [dispatch]);
  return (
    <div className={style.container} data-testid="details-container">
      <div className={style.top}>
        <img src={details.flag} alt="Flag" />
        <div className={style.name}>
          <h2>{details.name}</h2>
          <span>{details.timezone}</span>
        </div>

      </div>

      <div className={style.detail}>
        <span>Capital</span>
        <div>
          <span>{details.capital}</span>
          <img src={arrowImg} alt="arrow" />
        </div>
      </div>
      <div className={style.detail}>
        <span>Timezone</span>
        <div>
          <span>{details.timezone}</span>
          <img src={arrowImg} alt="arrow" />
        </div>
      </div>
      <div className={style.detail}>
        <span>population</span>
        <div>
          <span>{details.population}</span>
          <img src={arrowImg} alt="arrow" />
        </div>
      </div>
      <div className={style.detail}>
        <span>Formal Language</span>
        <div>
          <span>{details.firstLanguage}</span>
          <img src={arrowImg} alt="arrow" />
        </div>
      </div>
      <div className={style.detail}>
        <span>Area</span>
        <div>
          <span>{details.area}</span>
          <img src={arrowImg} alt="arrow" />
        </div>
      </div>
      <div className={style.detail}>
        <span>Currency</span>
        <div>
          <span>{details.currencies}</span>
          <img src={arrowImg} alt="arrow" />
        </div>
      </div>
    </div>
  );
}

export default Details;
