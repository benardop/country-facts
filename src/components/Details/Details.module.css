.container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: rgb(53, 84, 139);
  row-gap: 0;
  padding: 2rem  3rem;
}

.container img {
  width: 80%;
  max-width: 600px;
}

.container h2 {
  color: white;
  font-size: 2.5rem;
  font-weight: 700;
}

.container div {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.container div:nth-child(odd) {
  background-color: rgb(65, 104, 177);
}

.container div:nth-child(even) {
  background-color: rgb(63, 98, 166);
}

.container div span {
  font-size: large;
  color: white;
  flex-grow: 1;
  padding: 2rem;
  box-sizing: border-box;
}
