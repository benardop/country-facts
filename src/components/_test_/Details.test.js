import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Details from '../Details/Details';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component has changed', async () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Details />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check if the component container is there', async () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Details />
      </BrowserRouter>
    </Provider>,
  );
  const container = await screen.findByTestId('details-container');
  expect(container).toBeInTheDocument();
});
