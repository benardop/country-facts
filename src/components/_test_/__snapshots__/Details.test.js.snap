// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
<div
  className="container"
  data-testid="details-container"
>
  <div
    className="top"
  >
    <img
      alt="Flag"
    />
    <div
      className="name"
    >
      <h2 />
      <span />
    </div>
  </div>
  <div
    className="detail"
  >
    <span>
      Capital
    </span>
    <div>
      <span />
      <img
        alt="arrow"
        src="arrow.svg"
      />
    </div>
  </div>
  <div
    className="detail"
  >
    <span>
      Timezone
    </span>
    <div>
      <span />
      <img
        alt="arrow"
        src="arrow.svg"
      />
    </div>
  </div>
  <div
    className="detail"
  >
    <span>
      population
    </span>
    <div>
      <span />
      <img
        alt="arrow"
        src="arrow.svg"
      />
    </div>
  </div>
  <div
    className="detail"
  >
    <span>
      Formal Language
    </span>
    <div>
      <span />
      <img
        alt="arrow"
        src="arrow.svg"
      />
    </div>
  </div>
  <div
    className="detail"
  >
    <span>
      Area
    </span>
    <div>
      <span />
      <img
        alt="arrow"
        src="arrow.svg"
      />
    </div>
  </div>
  <div
    className="detail"
  >
    <span>
      Currency
    </span>
    <div>
      <span />
      <img
        alt="arrow"
        src="arrow.svg"
      />
    </div>
  </div>
</div>
`;
